pkg/scalers/selenium_grid_scaler.go

```diff
@@ -0,0 +1,223 @@
+	"k8s.io/metrics/pkg/apis/external_metrics"
+
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+)
+
+type seleniumGridScaler struct {
+	metadata *seleniumGridScalerMetadata
+	client   *http.Client
+}
+
+type seleniumGridScalerMetadata struct {
+	url            string
+	browserName    string
+	targetValue    int64
+	browserVersion string
+}
+
+type seleniumResponse struct {
+	Data data `json:"data"`
+}
+
+type data struct {
+	SessionsInfo sessionsInfo `json:"sessionsInfo"`
+}
+
+type sessionsInfo struct {
+	SessionQueueRequests []string          `json:"sessionQueueRequests"`
+	Sessions             []seleniumSession `json:"sessions"`
+}
+
+type seleniumSession struct {
+	ID           string `json:"id"`
+	Capabilities string `json:"capabilities"`
+	NodeID       string `json:"nodeId"`
+}
+
+type capability struct {
+	BrowserName    string `json:"browserName"`
+	BrowserVersion string `json:"browserVersion"`
+}
+
+const (
+	DefaultBrowserVersion string = "latest"
+)
+
+func NewSeleniumGridScaler(config *ScalerConfig) (Scaler, error) {
+	meta, err := parseSeleniumGridScalerMetadata(config)
+
+	if err != nil {
+		return nil, fmt.Errorf("error parsing selenium grid metadata: %s", err)
+	}
+
+	httpClient := kedautil.CreateHTTPClient(config.GlobalHTTPTimeout)
+
+	return &seleniumGridScaler{
+		metadata: meta,
+		client:   httpClient,
+	}, nil
+}
+
+func parseSeleniumGridScalerMetadata(config *ScalerConfig) (*seleniumGridScalerMetadata, error) {
+	meta := seleniumGridScalerMetadata{
+		targetValue: 1,
+	}
+
+	if val, ok := config.TriggerMetadata["url"]; ok {
+		meta.url = val
+	} else {
+		return nil, fmt.Errorf("no selenium grid url given in metadata")
+	}
+
+	if val, ok := config.TriggerMetadata["browserName"]; ok {
+		meta.browserName = val
+	} else {
+		return nil, fmt.Errorf("no browser name given in metadata")
+	}
+
+	if val, ok := config.TriggerMetadata["browserVersion"]; ok && val != "" {
+		meta.browserVersion = val
+	} else {
+		meta.browserVersion = DefaultBrowserVersion
+	}
+
+	return &meta, nil
+}
+
+// No cleanup required for selenium grid scaler
+func (s *seleniumGridScaler) Close() error {
+	return nil
+}
+
+func (s *seleniumGridScaler) GetMetrics(ctx context.Context, metricName string, metricSelector labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
+	v, err := s.getSessionsCount()
+	if err != nil {
+		return []external_metrics.ExternalMetricValue{}, fmt.Errorf("error requesting selenium grid endpoint: %s", err)
+	}
+
+	metric := external_metrics.ExternalMetricValue{
+		MetricName: metricName,
+		Value:      *v,
+		Timestamp:  metav1.Now(),
+	}
+
+	return append([]external_metrics.ExternalMetricValue{}, metric), nil
+}
+
+func (s *seleniumGridScaler) GetMetricSpecForScaling() []v2beta2.MetricSpec {
+	targetValue := resource.NewQuantity(s.metadata.targetValue, resource.DecimalSI)
+	metricName := kedautil.NormalizeString(fmt.Sprintf("%s-%s-%s", "seleniumgrid", s.metadata.browserName, s.metadata.browserVersion))
```

Comment:
might want to add `url` to make it truly unique

Reply:
Actually the value of url is going be to the same for all the scalers

Reply:
As discussed i have added the url to metric name to handle multiple grid scenario